protocol/signalfx/log_hecv1_json.go

```diff
@@ -0,0 +1,72 @@
+package signalfx
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/mailru/easyjson"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/logsink"
+	signalfxformat "github.com/signalfx/ingest-protocols/protocol/signalfx/format"
+)
+
+const (
+	// resources
+	host   = "host.hostname"
+	source = "service.name"
+	// attributes
+	sourceType = "source.type"
+	index      = "index"
+)
+
+// JSONDecoderHECV1 implements sink decoder for hecv1 splunk log events
+type JSONDecoderHECV1 struct {
+	Sink   logsink.Sink
+	Logger log.Logger
+}
+
+// Read implements interface method for decoding incoming json hecv1 format and convert it into golib/logsink.Log format
+func (decoder *JSONDecoderHECV1) Read(ctx context.Context, req *http.Request) (err error) {
+	var logHECV1List signalfxformat.JSONLogHECV1List
+
+	if err = easyjson.UnmarshalFromReader(req.Body, &logHECV1List); err == nil {
+		logs := make(logsink.Logs, 0, len(logHECV1List))
+		for _, lEntry := range logHECV1List {
+			if lEntry != nil {
+				e := &logsink.Log{
+					Body: lEntry.Event,
+					Resource: map[string]interface{}{
+						host:   lEntry.Host,
+						source: lEntry.Source,
+					},
+					Attributes: map[string]interface{}{
+						sourceType: lEntry.SourceType,
+						index:      lEntry.Index,
+					},
+				}
+
+				if lEntry.Time != nil {
+					e.TimeStamp = uint64(time.Duration(*lEntry.Time).Nanoseconds())
```

Comment:
Do you plan to keep lEntry.Time as `int64`? How are fractional seconds going to be handled?

Reply:
no i do not plan to keep it as int64. This is to get us started. Modifying the datamodel of HECv1 Wire Format will happen in subsequent commit

Reply:
@tigrannajaryan you are referring to `https://docs.splunk.com/Documentation/SplunkCloud/latest/Data/FormateventsforHTTPEventCollector#Event_metadata` ?

Reply:
Yes, in that link it says that "time" can be fractional seconds, with millisecond part. (Unclear if it will be a JSON number or JSON string).

Reply:
yes so i will handle that scenario in sub-sequent commits

Reply:
Sounds good.